week-4/implement/card-validator.test.js

```diff
@@ -0,0 +1,128 @@
+// In this project you'll write a script that validates whether or not a credit card number is valid.
+
+// Here are the rules for a valid number:
+
+// - Number must be 16 digits, all of them must be numbers.
+// - You must have at least two different digits represented (all of the digits cannot be the same).
+// - The final digit must be even.
+// - The sum of all the digits must be greater than 16.
+
+function isValidCardNumber(cardNum) {
+  if (
+    cardNum.length !== 16 ||
+    !Number(cardNum) ||
+    cardNum.slice(-1) % 2 !== 0
+  ) {
+    return false;
+  }
+  // Take first char
+  // Check if it is equal all other chars
+  // Count the equal numbers
+  // return false if count is equal to 16
+
+  let index = 0;
+  let firstChar = cardNum[index];
+  let count = 0;
+
+  while (index < cardNum.length) {
+    if (firstChar === cardNum[index]) {
+      count++;
+    }
+    index++;
+  }
+  if (count === 16) {
+    return false;
+  }
+
+  // Set total to 0
+  // Add all digits to total
+  // return false if count is less than 16
+
+  let total = 0;
+  index = 0;
+
+  while (index < cardNum.length) {
+    // Number conversion is needed otherwise "+" sign makes concatenation and give incorrect answer.
+    total = Number(cardNum[index]) + total;
+    index++;
+  }
+  if (total < 16) {
+    return false;
+  }
```

Comment:
Works perfectly and appreciate the comments here, afraid I have two of my own. 
- Don't use the same varible for index - it causes bugs something like this is better `while (let index ....)` This way each loop manages it's own index.
- There is a simpler way of doing this - genrally speaking when there's something like this that is often done, there will be (you'll develop a nose for this eventually)

I would use .split() to make the string into an array https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split and then call .reduce() on the array  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce